localstack-core/localstack/services/lambda_/provider.py

```diff
@@ -3524,8 +3525,10 @@ def publish_layer_version(
 
         layer = state.layers[layer_name]
         with layer.next_version_lock:
-            next_version = layer.next_version
-            layer.next_version += 1
+            next_version = LambdaLayerVersionIdentifier(
+                account_id=account, region=region, layer_name=layer_name
+            ).generate(next_version=layer.next_version)
+            layer.next_version = max(next_version, layer.next_version) + 1
```

Comment:
A comment might be helpful here to clarify the motivation of `max`, in particular to clarify when `next_version` != `layer.next_version`